extensions/git/src/git.ts

```diff
@@ -1791,13 +1791,21 @@ export class Repository {
 			.map(([ref]) => ({ name: ref, type: RefType.Head } as Branch));
 	}
 
-	async getRefs(opts?: { sort?: 'alphabetically' | 'committerdate', contains?: string }): Promise<Ref[]> {
+	async getRefs(opts?: { sort?: 'alphabetically' | 'committerdate', contains?: string, pattern?: string, count?: number }): Promise<Ref[]> {
 		const args = ['for-each-ref', '--format', '%(refname) %(objectname)'];
 
 		if (opts && opts.sort && opts.sort !== 'alphabetically') {
 			args.push('--sort', `-${opts.sort}`);
 		}
 
+		if (opts?.pattern) {
+			args.push(opts.pattern);
+		}
+
+		if (opts?.count) {
+			args.push(`--count=${opts.count}`);
+		}
+
 		if (opts?.contains) {
 			args.push('--contains', opts.contains);
 		}
```

Comment:
Let's make sure the order or args matches the order in the docs: https://git-scm.com/docs/git-for-each-ref#_synopsis

Reply:
Updated order of for-each-ref args to be count, sort, format, pattern, then contains.